src/components/NewPostImageUpload.tsx

```diff
@@ -38,14 +44,45 @@ const NewPostImageUpload = ({
     setUriList(deletedImageArray);
     onNewPostImageUpload(deletedImageArray);
   };
+
+  const getThumbnail = (uri: string): string | undefined => {
+    const isYoutubeVideo = uri.match(
+      /^https?\/\/(?:www\.youtube(?:nocookie)?\.com\/|m\.youtube\.com\/|youtube\.com\/)?(?:ytscreeningroom\?vi?=|youtu\.be\/|vi?\/|user\/.+\/u\/\w{1,2}\/|embed\/|watch\?(?:.*)?vi?=|vi?=|\?(?:.*)?vi?=)([^#\n/<>"']*)/i
```

Comment:
Are we using this (or a subset of this) anywhere else?  Might be good to extract it (and the other two patterns) into something reusable.

Reply:
we are not. we're checking but not filtering in the same way because we just care if it's a video in react player.  in this case, we need to know more specific details.